zerver/models.py

```diff
@@ -1941,19 +1941,21 @@ class UserHotspot(models.Model):
     class Meta:
         unique_together = ("user", "hotspot")
 
-def check_valid_user_id(realm_id: int, user_id: object,
-                        allow_deactivated: bool=False) -> Optional[str]:
-    if not isinstance(user_id, int):
-        return _("User id is not an integer")
+def check_valid_user_ids(realm_id: int, user_ids: List[object],
+                         allow_deactivated: bool=False) -> Optional[str]:
+    error = check_list(check_int)("User IDs", user_ids)
+    if error:
+        return error
     try:
-        realm = Realm.objects.get(id=realm_id)
-        user_profile = get_user_profile_by_id_in_realm(user_id, realm)
-        if not allow_deactivated:
-            if not user_profile.is_active:
-                return _('User is deactivated')
-
-        if (user_profile.is_bot):
-            return _('User with id %d is bot') % (user_id)
+        for user_id in user_ids:
+            realm = Realm.objects.get(id=realm_id)
+            user_profile = get_user_profile_by_id_in_realm(user_id, realm)
+            if not allow_deactivated:
```

Comment:
Even if we have checked user_id type in check_int validation, mypy throws error here, as function `get_user_profile_by_id_in_realm` takes integer type of argument and user_id is object. To remove this error we have explicitly again add a check for integer value before calling function. Is there any other solution to solve this?